tests/integration/semantics/src/test/java/org/apache/pulsar/tests/integration/functions/PulsarFunctionsTestBase.java

```diff
@@ -50,8 +50,8 @@ protected static String getExclamationClass(Runtime runtime) {
     }
 
     @BeforeClass
-    public static void setupCluster() throws Exception {
-        PulsarClusterTestBase.setupCluster();
+    public void setupCluster() throws Exception {
```

Comment:
why do you need to remove 'static' here? All tests here share same cluster, and they all annotated with `BeforeClass`.

Reply:
you cannot they the instantiated class instance when the method is static.

Reply:
what do you mean they can instantiate class instance? when you switch from static method to class method, you are setting up and tearing down clusters for each test method, which is not necessary. unless there is a strong reason to do that, I would prefer not changing that.

Reply:
No we are not doing that that happens with @BeforeMethod and @AfterMethod

Reply:
fine, but why do you change it from static method to class method?

if you do have a strong reason, please describe the reason in your pull request. otherwise it is hard for reviewers to understand why these are changed.

Reply:
it won't compile otherwise the base class method is no longer static

Reply:
fine. please update the PR description describe the motivation and changes. like what other PRs are doing:

https://github.com/apache/incubator-pulsar/pull/2079